mono/metadata/w32file-unix.c

```diff
@@ -4329,20 +4333,25 @@ mono_w32file_get_disk_free_space (const gunichar2 *path_name, guint64 *free_byte
 	gint ret;
 	unsigned long block_size;
 
+	*win32error = 0;
+	// FIXME check for embedded nuls in native or managed
+
 	if (path_name == NULL) {
 		utf8_path_name = g_strdup (g_get_current_dir());
 		if (utf8_path_name == NULL) {
-			mono_w32error_set_last (ERROR_DIRECTORY);
-			return(FALSE);
+			mono_w32error_set_last (ERROR_DIRECTORY); // FIXME remove this
```

Comment:
These are just plain duplication for no useful reasons, and even then, the change is just plain wrong. There is a reason for using a TLS variable, and changing without understanding is not the way to go :wink:. Revert all these changes related to using `win32error` local variable.

Reply:
TLS is kinda a hack that this works toward removing. It is preferable to use parameters, like we often do.

Reply:
Instead of dismissing it as hack, did you try to understand first why it’s here? I already pointed in the previous comment that it’s fulfilling another purpose than “just” passing it as a local variable.